build/build-scripts.sh

```diff
@@ -0,0 +1,109 @@
+
+VENV_DIR=${HERE}/venv
+mkdir VENV_DIR
+export VIRTUAL_ENV=$VENV_DIR
+python3 -m venv $VIRTUAL_ENV
+export PATH="$VIRTUAL_ENV/bin:$PATH"
+
+pip install wheel
+pip install pytest>3.6 \
+            pytest-cov \
+	    coveralls \
+	    coverage~=4.5.4 \
+            urllib3==1.23 \
+	    mock
+
+# Install pygsl
+sudo apt-get install -y gsl-bin libgsl-dev libgsl-dbg
+
+# Install RALFit
+sudo apt-get install -y gfortran lcov libblas-dev liblapack-dev libgdal-dev
+git clone https://github.com/ralna/RALFit
+pip install numpy
+mkdir -p ${HERE}/RALFit/libRALFit/build
+cd ${HERE}/RALFit/libRALFit/build
+cmake .. && make && make install
+pip install .
+export LD_LIBRARY_PATH=${HERE}/RALFit/libRALFit/build/src:$LD_LIBRARY_PATH
+
+# Install  cutest
+CUTEST_DIR=${HERE}/cutest
+mkdir -p ${CUTEST_DIR}
+cd ${CUTEST_DIR}
+git clone https://github.com/ralna/ARCHDefs ./archdefs 
+git clone https://github.com/ralna/SIFDecode ./sifdecode 
+git clone https://github.com/ralna/CUTEst ./cutest
+mkdir pycutest_cache
+export ARCHDEFS=${CUTEST_DIR}/archdefs/ 
+export SIFDECODE=${CUTEST_DIR}/sifdecode/
+export MASTSIF=${HERE}/fitbenchmarking/examples/benchmark_problems/SIF/ 
+export CUTEST=${CUTEST_DIR}/cutest/ 
+export MYARCH="pc64.lnx.gfo" 
+export PYCUTEST_CACHE=${CUTEST_DIR}/pycutest_cache/
+export PYTHONPATH="${PYCUTEST_CACHE}:${PYTHONPATH}"
+cd $SIFDECODE
+printf "6\n2\n6\n" > sifdecode.input 
+printf "nny" >> sifdecode.input
+./install_sifdecode  < sifdecode.input
+cd $CUTEST
+printf "6\n2\n6\n2\n8\n" > cutest.input
+printf "nnydn" >> cutest.input
+./install_cutest < cutest.input
+pip install pycutest
+
+# Install mantid
+cd $HERE
+# set noninteractive to stop tzdata prompt
+export DEBIAN_FRONTEND=noniteractive
+OLD_VENV=$VIRTUAL_ENV
+export VIRTUAL_ENV=""
+export OLD_PATH=$PATH
+export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+sudo apt-get install -y git g++ qt5-default clang-format-6.0 cmake dvipng doxygen libtbb-dev \
+  libgoogle-perftools-dev libboost-all-dev libpoco-dev libnexus-dev libhdf5-dev libhdf4-dev \
+  libjemalloc-dev libgsl-dev liboce-visualization-dev libmuparser-dev libssl-dev libjsoncpp-dev \
+  librdkafka-dev qtbase5-dev qttools5-dev qttools5-dev-tools libqt5webkit5-dev \
+  libqt5x11extras5-dev libqt5opengl5-dev libqscintilla2-qt5-dev libpython3-dev \
+  ninja-build python3-setuptools python3-sip-dev python3-pyqt5 pyqt5-dev pyqt5-dev-tools \
+  python3-qtpy python3-numpy python3-scipy python3-sphinx python3-sphinx-bootstrap-theme \
+  python3-pycifrw python3-dateutil python3-matplotlib python3-qtconsole python3-h5py \
+  python3-mock python3-psutil python3-requests python3-toml python3-yaml wget
+
+# Details from mantid website
+wget https://downloads.sourceforge.net/project/mantid/6.3/mantid-6.3.0-Source.tar.xz
+tar -xvf mantid-6.3.0-Source.tar.xz
+rm mantid*.xz
+
+sudo apt-get install -y ccache
+ccache --max-size=20G
+
+mkdir -p Mantid
+
+cd Mantid
+cmake ${HERE}/mantid-6.3.0-Source -DENABLE_MANTIDPLOT=OFF
+make
```

Comment:
This is what I did to get a working 6.2.0 -- I think it's just the `DENABLE_PRECOMMIT` that's important though. Also Sarah mentioned that Mantid is on 6.5.0 now??

We also found that numpy-1.21.6 works but numpy-1.23.4 does not for mantid-6.2.0
```
wget https://downloads.sourceforge.net/project/mantid/6.2/mantid-6.2.0-Source.tar.xz
tar -xvf mantid-6.2.0-Source.tar.xz
sudo mkdir /opt/Mantid
cd /opt/Mantid-6.2.0/
sudo cmake ~/mantid-6.2.0-Source/ -DENABLE_PRECOMMIT=OFF
sudo make
```

Reply:
@AndrewLister-STFC yes Mantid is on 6.5.0 but they now provide a prebuilt binary that should just work (no longer a need to compile from source), see https://www.mantidproject.org/installation/index.html#installation-full

Reply:
🙃 they used to do this and then stopped... hence why we had to do all this building

Reply:
I've also now got v6.6.0 running on python 3.10 using the above steps. I found that I had to remove the `qt5-default` and `clang-format-6.0` packages from the `sudo apt-get install` command and then also add `libeigen3-dev`. I also needed to pip install `versioningit`